### ISSUE TITLE
Project Panama

### ISSUE THREAD

**Issue by pschichtel** [pschichtel/JavaCAN#38]
It would be nice to have a version of JavaCAN that uses Project Panama for SocketCAN and epoll interaction instead of JNI, making this a pure Java project.

jextract might come in handy for generating the low-level glue-code:
http://cr.openjdk.java.net/~mcimadamore/panama/jextract_distilled.html

**Comment by splatch** [comment]
BTW have you thought of giving a try to jnr-ffi or JNA? I think both allow to cut JNI part with some performance sacrifices. I saw it used with dbus (for unix sockets) and systemd (to call its library). Both work with present releases of JVM.

**Comment by pschichtel** [comment]
Yes, both would work but face the exact same challenge as project panama.

**Comment by pschichtel** [comment]
labeling this for the next major version. I don't think it makes sense to do this without rethinking some API decisions.

**Comment by pschichtel** [comment]
Java 22 makes this Project Panama stable: https://openjdk.org/projects/jdk/22/

**Comment by pschichtel** [comment]
Java 25 (first LTS to include panama) will be released this september, that might be time to revisit this.
